tests/benchmark/read_gbq_colab/first_page.py

```diff
@@ -27,8 +27,9 @@ def first_page(*, project_id, dataset_id, table_id):
         f"SELECT * FROM `{project_id}`.{dataset_id}.{table_id}"
     )
 
-    # Get number of rows (to calculate number of pages) and the first page.
-    df.shape
+    # Use total_rows from batches directly and the first page
+    execute_result = df._block.session._executor.execute(df._block.expr, ordered=True)
+    execute_result.total_rows or 0
     next(iter(df.to_pandas_batches(page_size=PAGE_SIZE)))
```

Comment:
I think this would cause the query to execute twice. Please revert. Instead, it should match as closely as possible to what happens in TableWidget.

```suggestion
    # Get number of rows (to calculate number of pages) and the first page.
    batches = df.to_pandas_batches(page_size=PAGE_SIZE)
    first_page = next(iter(batches))
    assert first_page is not None
    total_rows = batches.total_rows
    assert total_rows is not None
```